### PR TITLE
Update models to LWM2M 1.1

### PR DESCRIPTION
**Pull request by sbernard31**
Update ddf files and all parser/validator.

### CODE REVIEW

**Comment by dachaac** [comment]
I was updating certificate support a bit in here:
https://github.com/dachaac/leshan/commits/improve-certificate-support

Mainly commits:
https://github.com/dachaac/leshan/commit/b7a881504ee3b43d99a1cf7b5779c9159601cd19 - schemas: Add schema for LWM2M 1.1
https://github.com/dachaac/leshan/commit/a0baa3054d015298dc87c66c0ac9d31a4b00d33b - ResourceModel: Add new type "Unsigned Integer" from LWM2M 1.1
https://github.com/dachaac/leshan/commit/77b36d3ebfbf51a4506d09fb6cfae331f172983c - LWM2M_Security-v1_1_1.xml: Add new Security object definition from LWM2M 1.1.1

My changes weren't that complete as what you have here but I also did changes to other stuff in order to handle new fields in Security object in later commits in that branch.

Do you want to merge part of that stuff in here or I will do the updating PR? Mainly the "Unsigned Integer" support is what I am interested in. 

Others changes there were just to get new Security Object in use. It may be that we don't need those other commits after your changes.

**Comment by sbernard31** [comment]
I finished what I planned for this PR.

Most of your changes are present in this PR, except adding support for **Unsigned Integer**.
I think this should be done in another PR.

But currently my feeling is that this will not be so easy, because the specification defines **unsigned integer** as  
_"An 8, 16, 32 or 64-bit unsigned integer."_ (source : [Appendix-C-Data-Types](http://www.openmobilealliance.org/release/LightweightM2M/V1_1-20180710-A/HTML-Version/OMA-TS-LightweightM2M_Core-V1_1-20180710-A.html#11-0-Appendix-C-Data-Types-Normative)).

Generally in java, the easy way to support unsigned int is to use a primitive type with a larger size but we can not do that because we need to support  64-bit unsigned integer.

A solution could be to use `BigInteger` or create/reuse dedicated classes (e.g. [`UnsignedLong`](https://guava.dev/releases/snapshot/api/docs/com/google/common/primitives/UnsignedLongs.html)) which store the embedded value in a long.

I clearly need to think a bit more about that. If you have any opinion on this please share it.
I should also discuss about that with @msangoi.

**Comment by dachaac** [comment]
Didn't spot anything wrong with the changes in here. Once the Unsigned Integer support is merged with forthcoming PR then I will rebase my code on top of that.

Unfortunately Java doesn't have the unsigned type -- don't really have opinion of what class to use to handle full range. BigInteger might be a bit slow thou. SecurityObject's usage would easily fit even in 8 bits ;). But serialization from different formats might be the thing to consider. At least Jackson has support for BigInteger so perhaps that would be the easiest.

File: leshan-core/src/main/java/org/eclipse/leshan/core/model/DDFFileParser.java [hunk 1]
```diff
@@ -89,20 +105,32 @@ public DDFFileParser(DDFFileValidator ddfValidator) {
...
+            if (ddfFileValidator != null) {
+                ddfFileValidator.validate(document);
             }
 
             // Build list of ObjectModel
             ArrayList<ObjectModel> objects = new ArrayList<>();
             NodeList nodeList = document.getDocumentElement().getElementsByTagName("Object");
             for (int i = 0; i < nodeList.getLength(); i++) {
-                objects.add(parseObject(nodeList.item(i), streamName, ddfValidator != null));
+                objects.add(parseObject(nodeList.item(i), streamName, lwm2mVersion, ddfValidatorFactory != null));
```
[the remark this answers is not in the document]
Comment (sbernard31) [reply leshan-core/src/main/java/org/eclipse/leshan/core/model/DDFFileParser.java:133]: Good catch !

File: leshan-core/src/main/java/org/eclipse/leshan/core/model/DDFFileValidatorFactory.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    /**
+     * Extract LWM2M version from DDF file
+     * 
+     * @param document the DDF file.
+     * @param documentName the DDF file name for error raising or logging purpose.
+     * 
+     * @return the version of the minimal LWM2M version supported by this DDF file.
+     */
+    Version extractLWM2MVersion(Document document, String documentName) throws InvalidDDFFileException;
```
[the remark this answers is not in the document]
Comment (sbernard31) [reply leshan-core/src/main/java/org/eclipse/leshan/core/model/DDFFileValidatorFactory.java:34]: The factory is able to create a validator for a given version.
But as there is not "standard" way to extract the version, I prefer to delegate it to the factory.

**Pull request merged**